pytorch_lightning/trainer/configuration_validator.py

```diff
@@ -290,3 +292,14 @@ def _check_dl_idx_in_on_train_batch_hooks(trainer: "pl.Trainer", model: "pl.Ligh
                     f"Base `Callback.{hook}` hook signature has changed in v1.5."
                     " The `dataloader_idx` argument will be removed in v1.7."
                 )
+
+
+def _check_on_init_start_end(trainer: "pl.Trainer") -> None:
+    """Checks if on_init_start/end are overridden and sends a deprecation warning."""
+    for callback in trainer.callbacks:
+        if is_overridden(method_name="on_init_start", instance=callback):
+            rank_zero_deprecation(
+                "The `on_init_start` callback hook was deprecated in v1.6 and will be removed in v1.8."
```

Comment:
Should we mention to use setup() instead?

Reply:
IMO we shouldn't because that's up to the user to look for another hook since the `Callback.setup` is called after the environment is already being setup up and we can't say for sure what all crazy stuff user might be doing inside `on_init` functions which could fail if they go for `setup`.

Reply:
Sure, added

Reply:
oh sorry just saw your comment Rohit. I actually agree. Will put it back to how it was before